src/lib/redirectUserTo.ts

```diff
@@ -4,7 +4,7 @@ export const redirectUserTo = (
   url: string
 ): GetServerSidePropsResult<null> => ({
   redirect: {
+    statusCode: 303,
     destination: url,
-    permanent: false,
   },
```

Comment:
Specifying a [303 status code](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/303) and removing 'permanent' (which Next turns into a 307) turns a POST request into a GET request.  This means the `Rule`s can make correct assertions about what the user is doing when they are re-directed after submitting a valid form.  

Without the 303 status code, the POSTing of a valid form results in the redirected request to the next page also being a POST request.